girder_worker/docker/transforms/girder.py

```diff
@@ -47,22 +47,23 @@ def transform(self, **kwargs):
 
 
 class GirderFileIdToVolume(GirderClientTransform):
-    def __init__(self, _id, volume=TemporaryVolume.default, **kwargs):
+    def __init__(self, _id, volume=TemporaryVolume.default, filename=None, **kwargs):
```

Comment:
Are we happy with the user having to pass the filename in when we could extract if from Girder?

Reply:
I considered that as another option. Ideally both would be possible. Maybe another option `use_girder_filename`?

Reply:
I was thinking that if filename was ```None``` we could get it from Girder?

Reply:
I'd be OK with that, though it breaks the existing behavior.

Reply:
I guess one issue with that approach is filename conflicts, where as the default of using the uuid is going to be pretty safe from that point of view.

Reply:
Right.

Reply:
I guess we could get the best of both worlds with <root>/<file_id>/<filename>?

Reply:
I like that plan, it also mitigates the risk of any output paths conflicting with input paths.

I'll do that extra directory as the default case, but if a filename is passed explicitly, we'll leave out the UUID. That will allow clients max control, which may be useful in odd cases like where two files need to be in the same directory (e.g. mha/mhd read side effects).

Reply:
Sounds good.